File: 560. Subarray Sum Equals K/subarraySum.py
class subarraySum:
    def solution(self, nums: List[int], k: int) -> int:
        sum = 0
        res = 0
        cul = {}
        cul[0] = 1
        for n in nums:
            sum += n
            if cul.__contains__(sum - k):
                res += cul[sum - k]
            if not cul.__contains__(sum):
                cul[sum] = 0
            cul[sum] += 1
        return res
            

File: 560. Subarray Sum Equals K/subarraySum.java
class subarraySum {
    public int solution(int[] nums, int k) {
        int sum = 0, res = 0;
        Map<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);
        for (int n : nums) {
            sum += n;
            if (map.containsKey(sum - k)) {
                res += map.get(sum - k);
            }
            map.put(sum, map.getOrDefault(sum, 0) + 1);
        }
        return res;
    }
}